src/transformers/trainer.py

```diff
@@ -1896,6 +1899,18 @@ def _inner_training_loop(
                         if not isinstance(self.lr_scheduler, torch.optim.lr_scheduler.ReduceLROnPlateau):
                             self.lr_scheduler.step()
 
+
+                    xm.mark_step()
+                    if self.num_compilations != met.metric_data('CompileTime')[:1] :
+                       self.num_compilations = met.metric_data('CompileTime')[:1]
+                    else:
+                       xm.rendezvous('step')
+                       step_time = time.time() - self.last_time_stamp
+                       data, fsdp, mdl = self.args.spmd_mesh.ici_mesh_shape
+                       num_devices = data * fsdp * mdl
+                       num_tokens = inputs["input_ids"].numel() / num_devices
+                       xm.master_print(f"Step time: {step_time}: Model TFLOPS: {self.model_flops(step_time, num_tokens)}")
```

Comment:
cc @yeounoh, we can emit this to Tensorboard for perf tests (in addition to the convergence metrics)